src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -50,32 +50,302 @@ Notifier::~Notifier (void)
+    headers[key] = value;
+  }
+
+
+  //
+  // 6. Split URI in protocol, host, port and path
+  //
+  std::string  protocol;
+  std::string  host;
+  int          port;
+  std::string  uriPath;
+
+  if (!parseUrl(url, host, port, uriPath, protocol))
+  {
+    LM_E(("Runtime Error (not sending NotifyContextRequest: malformed URL: '%s')", httpInfo.url.c_str()));
+    return false;
+  }
+
+
+  //
+  // 7. Add URI params from template to uriPath
+  //
+  std::string  uri = uriPath;
+
+  if (headers.size() != 0)
+  {
+    uri += "?";
+
+    int ix = 0;
+    for (std::map<std::string, std::string>::iterator it = qs.begin(); it != qs.end(); ++it)
+    {
+      if (ix != 0)
+      {
+        uri += "&";
+      }
+
+      uri += it->first + "=" + it->second;
+      ++ix;
+    }
+  }
+
+
+  //
+  // 8. HTTP headers - to httpRequestSend via parameter 'extraHeaders'
+  //
+  std::string  out;
+  int          r;
+
+  LM_W(("KZ: Sending notification: %s [%s]", uri.c_str(), payload.c_str()));
+  r = httpRequestSend(host,
+                      port,
+                      protocol,
+                      method,
+                      tenant,
+                      ce.entityId.servicePath,
+                      xauthToken,
+                      uri,
+                      "application/json",
+                      payload,
+                      fiwareCorrelator,
+                      "",                  // ngisv2AttrFormat
+                      false,               // don't use rush ... ?
+                      true,                // wait for response
+                      &out,
+                      headers,
+                      "application/json",  // Accept Format
+                      5000);               // Timeout in milliseconds: FIXME PR: unhardwire
+
+  if (r == 0)
+  {
+    statisticsUpdate(NotifyContextSent, JSON);
+    alarmMgr.notificationErrorReset(url);
+    return true;
+  }
+
+  return false;
+}
+
+
+
+/* ****************************************************************************
+*
+* NotificationAsTemplateParams - 
+*/
+typedef struct NotificationAsTemplateParams
+{
+  NotifyContextRequest*            ncrP;
+  ngsiv2::HttpInfo                 httpInfo;
+  std::string                      tenant;
+  std::string                      xauthToken;
+  std::string                      fiwareCorrelator;
+  RenderFormat                     renderFormat;
+  std::vector<std::string>         attrsOrder;
+} NotificationAsTemplateParams;
+
+
+
+/* ****************************************************************************
+*
+* sendNotifyContextRequestAsPerTemplate -
+*/
+void* sendNotifyContextRequestAsPerTemplate(void* p)
+{
+  NotificationAsTemplateParams* paramP = (NotificationAsTemplateParams*) p;
+
+  LM_W(("KZ: In sendNotifyContextRequestAsPerTemplate (%d contextElementResponses)", paramP->ncrP->contextElementResponseVector.size()));
+
+  for (unsigned int ix = 0; ix < paramP->ncrP->contextElementResponseVector.size(); ++ix)
```

Comment:
This is the part in which you split into N HTTP requests in the case of templates?